google/cloud/pubsub/samples/samples.cc

```diff
@@ -221,6 +221,109 @@ void Subscribe(google::cloud::pubsub::Subscriber subscriber,
   (std::move(subscriber), std::move(subscription));
 }
 
+void CustomThreadPoolPublisher(std::vector<std::string> const& argv) {
+  namespace examples = ::google::cloud::testing_util;
+  if (argv.size() != 2 || (argv.size() == 1 && argv[0] == "--help")) {
```

Comment:
Given that `lhs` is false when evaluating `rhs` in `lhs || rhs`, and that `!lhs` implies `!rhs` in this case, this guard is equivalent to:
```
if (argv.size() != 2) {
```
You might say the existing incantation is a more general pattern that allows for `s/2/1/` should the command only take one argument, but, if such an edit was made, I would probably then suggest changing
```
if (argv.size() != 1 || (argv.size() == 1 && argv[0] == "--help")) {
```
to just
```
if (argv.size() != 1 || argv[0] == "--help") {
```
In either case, simplicity should win, methinks.

Reply:
Doh! Fixed.